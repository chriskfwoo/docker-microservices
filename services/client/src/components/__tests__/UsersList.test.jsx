import React from 'React'
import { shallow } from 'enzyme'
import UsersList from '../UsersList'
import renderer from 'react-test-renderer'

const users = [
  {
    'active': true,
    'email': 'chris@gmail.com',
    'id': 1,
    'username': 'chriswoo'
  },
  {
    'active': true,
    'email': 'johndoe@gmail.com',
    'id': 2,
    'username': 'johndoe'
  }
];

test('UserList renders properly', () => {
  const wrapper = shallow(<UsersList users={users} />);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('chriswoo')
});

test('UserList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});