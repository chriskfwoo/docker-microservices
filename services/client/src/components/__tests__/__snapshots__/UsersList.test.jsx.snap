// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    chriswoo
  </h4>
  <h4
    className="box title is-4"
  >
    johndoe
  </h4>
</div>
`;
